presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -589,7 +595,11 @@ private S3Object getS3Object(final Path path, final long start)
                                 return s3.getObject(new GetObjectRequest(host, keyFromPath(path)).withRange(start, Long.MAX_VALUE));
                             }
                             catch (AmazonServiceException e) {
-                                if (e.getStatusCode() == SC_FORBIDDEN) {
+                                if (e.getStatusCode() == SC_REQUESTED_RANGE_NOT_SATISFIABLE) {
+                                    log.info("The given starting byte(%d) was past the end of file(%s), ignore it. (%s)", start, path, e.getMessage());
+                                    return new S3Object();
+                                }
+                                else if (e.getStatusCode() == SC_FORBIDDEN) {
```

Comment:
No need to add an `else` here since the above `if` always returns (these `else` is redundant`).